lib/src/modules/macho/mod.rs

```diff
@@ -446,6 +446,38 @@ fn import_hash(ctx: &mut ScanContext) -> Option<RuntimeString> {
     Some(RuntimeString::new(digest))
 }
 
+/// Returns an md5 hash of the symbol table in the mach-o binary
+#[module_export]
+fn sym_hash(ctx: &mut ScanContext) -> Option<RuntimeString> {
+    let macho = ctx.module_output::<Macho>()?;
+    let mut symtab_to_hash = &macho.symtab.entries;
+
+    // if there is not a symbtol table in the main Macho, the symbol table of the
+    // nested file should be hashed
+    if symtab_to_hash.is_empty() && !macho.file.is_empty() {
+        symtab_to_hash = &macho.file[0].symtab.entries;
+    }
+
+    // we need to check again as the nested symbol table could be empty too
+    if symtab_to_hash.is_empty() {
+        return None;
+    }
+
+    let mut md5_hash: digest::core_api::CoreWrapper<md5::Md5Core> = Md5::new();
+
+    let symtab_hash_entries = symtab_to_hash
+        .iter()
+        .map(|e| e.trim().to_lowercase().as_bstr().to_owned())
```

Comment:
Here it's more efficient to do `BString::new(e.trim().to_lowercase())`. `to_lowercase` returns a `Vec<u8>`, when you do `.as_bstr().to_owned()`, you are basically cloning the vector and wrapping it in a `BString`. By using `BString::new` you are given the ownership of the original vector to `BString`, and no clone is performed.

Reply:
Ahhh, I see now. Thanks for pointing that out Victor, appreciated 😄